auth.go

```diff
@@ -13,27 +16,54 @@ type AuthenticationResponse struct {
 	Scope        string `json:"scope"`
 }
 
-// Authenticate sends an authentication request, and returns a newly constructed client object
+var cache = ttlcache.New[string, string]()
+
+// Authenticate sends an authentication request when a token is not available in the cache, and returns a newly constructed client object
 func Authenticate(username, password, apiEndpoint string) (*Client, error) {
 	infolog.Printf("Start authentication into '%s' using '%s'", apiEndpoint, LoginUrl)
+	userAgent := ConstructUserAgentString()
+	cachedToken := getCachedAccessToken(username, apiEndpoint)
+	if len(cachedToken) > 0 {
+		infolog.Printf("Returning auth token from cache")
```

Comment:
maybe this is better at a debug level? I feel like this is too much info for info level (pardon my info).

Reply:
Yep I will change that ! I don't think it's even needed